solr/core/src/java/org/apache/solr/cloud/api/collections/ReindexCollectionCmd.java

```diff
@@ -103,17 +104,19 @@ public class ReindexCollectionCmd implements CollApiCmds.CollectionApiCommand {
   public static final String STATE = "state";
   public static final String PHASE = "phase";
 
-  private static final List<String> COLLECTION_PARAMS =
-      Arrays.asList(
-          ZkStateReader.CONFIGNAME_PROP,
-          ZkStateReader.NUM_SHARDS_PROP,
-          ZkStateReader.NRT_REPLICAS,
-          ZkStateReader.PULL_REPLICAS,
-          ZkStateReader.TLOG_REPLICAS,
-          ZkStateReader.REPLICATION_FACTOR,
-          "shards",
-          CollectionAdminParams.CREATE_NODE_SET_PARAM,
-          CollectionAdminParams.CREATE_NODE_SET_SHUFFLE_PARAM);
+  private static final List<String> COLLECTION_PARAMS = makeCollectionParams();
+
+  private static List<String> makeCollectionParams() {
+    List<String> collectionParams = new ArrayList<>(10);
+    collectionParams.add(ZkStateReader.CONFIGNAME_PROP);
+    collectionParams.add(ZkStateReader.NUM_SHARDS_PROP);
+    collectionParams.add(ZkStateReader.REPLICATION_FACTOR);
+    collectionParams.add("shards");
+    collectionParams.add(CollectionAdminParams.CREATE_NODE_SET_PARAM);
+    collectionParams.add(CollectionAdminParams.CREATE_NODE_SET_SHUFFLE_PARAM);
+    collectionParams.addAll(CollectionHandlingUtils.numReplicasProperties());
+    return List.copyOf(collectionParams);
```

Comment:
Same as in a similar comment in another class: to make it immutable. If Guava was available it would build an ImmutableList in one shot, but if I understood well the goal was be to remove Guava.

Reply:
Removed that one copy as well, since it is purely internal anyway.